manager/controllers/app/data_catalog_interface.go

```diff
@@ -22,44 +20,6 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-// DataCatalog is an interface of a facade to a data catalog.
-type DataCatalog interface {
-	pb.DataCatalogServiceServer
-	io.Closer
-}
-
-type DataCatalogImpl struct {
-	catalogClient     pb.DataCatalogServiceClient
-	catalogConnection *grpc.ClientConn
-}
-
-func NewGrpcDataCatalog() (*DataCatalogImpl, error) {
-	catalogConnection, err := grpc.Dial(utils.GetDataCatalogServiceAddress(), grpc.WithInsecure(), grpc.WithBlock())
-	if err != nil {
-		return nil, err
-	}
-
-	return &DataCatalogImpl{
-		catalogClient:     pb.NewDataCatalogServiceClient(catalogConnection),
-		catalogConnection: catalogConnection,
-	}, nil
-}
-
-func (d *DataCatalogImpl) GetDatasetInfo(ctx context.Context, req *pb.CatalogDatasetRequest) (*pb.CatalogDatasetInfo, error) {
-	result, err := d.catalogClient.GetDatasetInfo(ctx, req)
-	return result, errors.Wrap(err, "get dataset info failed")
-}
-
-func (d *DataCatalogImpl) RegisterDatasetInfo(ctx context.Context, req *pb.RegisterAssetRequest) (*pb.RegisterAssetResponse, error) {
-	result, err := d.catalogClient.RegisterDatasetInfo(ctx, req)
-	return result, errors.Wrap(err, "register dataset info failed")
-}
-
-func (d *DataCatalogImpl) Close() error {
-	err := d.catalogConnection.Close()
-	return err
-}
-
 // RegisterAsset registers a new asset in the specified catalog
```

Comment:
Seems as the data catalog facade is not used in
https://github.com/mesh-for-data/mesh-for-data/blob/e01317957492ba978efd922c1dd60cd645ba102c/manager/controllers/app/data_catalog_interface.go#L32

Reply:
Changed